### DISCUSSION TITLE
How to enable moving files to trash?

### DISCUSSION THREAD

**Discussion by emanueleperuffo** [coder/code-server#4469] [Ideas]
I have tried to use the ELECTRON_TRASH environment variables:
- with glib installed and ELECTRON_TRASH set to gio
- with trash-cli installed and ELECTRON_TRASH set to trash-cli

Unfortunately, I did not see the move to trash option enabled, only the delete permanently option.

I'm not much familiar with the vscode source, but I saw after some searching in github that the is some checking for the platform vscode is running in.

Someone knows how to address this? I think it would add great value to code-server.

**Comment by lg-nineoneninethreenine** [comment]
Hmm... I think @jsjoeio or @code-asher might help on this...

**Reply by jsjoeio** [reply]
I don't think we support these 🤔 cc @code-asher

**Reply by code-asher** [reply]
Yeah it appears only the Electron version supports the trash, at least for now.  Here is the Electron file service which is the same as the base service except it adds trash support:

https://github.com/cdr/vscode/blob/d62e8db202f80db7a42233cd56d04e6806109fb1/src/vs/workbench/services/files/electron-browser/diskFileSystemProvider.ts#L26

We could probably do something similar except call out to some Node module to handle the trash.

**Reply by jsjoeio** [reply]
Oh nice! Opened an issue and put under Backlog Candidates here: https://github.com/cdr/code-server/issues/4486
